src/Avalonia.Controls/SplitView.cs

```diff
@@ -0,0 +1,476 @@
+                IsPaneOpen = false;
+        }
+
+        private bool ShouldClosePane()
+        {
+            return (DisplayMode == SplitViewDisplayMode.CompactOverlay || DisplayMode == SplitViewDisplayMode.Overlay);
+        }
+
+        protected virtual void OnPaneOpening(SplitView sender, EventArgs args)
+        {
+            PaneOpening?.Invoke(sender, args);
+        }
+
+        protected virtual void OnPaneOpened(SplitView sender, EventArgs args)
+        {
+            PaneOpened?.Invoke(sender, args);
+        }
+
+        protected virtual void OnPaneClosing(SplitView sender, SplitViewPaneClosingEventArgs args)
+        {
+            PaneClosing?.Invoke(sender, args);
+        }
+
+        protected virtual void OnPaneClosed(SplitView sender, EventArgs args)
+        {
+            PaneClosed?.Invoke(sender, args);
+        }
+
+        private void OnCompactPaneLengthChanged(AvaloniaPropertyChangedEventArgs e)
+        {
+            var newLen = (double)e.NewValue;
+            var displayMode = DisplayMode;
+            if (displayMode == SplitViewDisplayMode.CompactInline)
+            {
+                TemplateSettings.ClosedPaneWidth = newLen;
+            }
+            else if (displayMode == SplitViewDisplayMode.CompactOverlay)
+            {
+                TemplateSettings.ClosedPaneWidth = newLen;
+                TemplateSettings.PaneColumnGridLength = new GridLength(newLen, GridUnitType.Pixel);
+            }
+        }
+
+        private void OnPanePlacementChanged(AvaloniaPropertyChangedEventArgs e)
+        {
+            var oldState = e.OldValue.ToString().ToLower();
+            var newState = e.NewValue.ToString().ToLower();
+            PseudoClasses.Remove($":{oldState}");
+            PseudoClasses.Add($":{newState}");
+        }
+
+        private void OnDisplayModeChanged(AvaloniaPropertyChangedEventArgs e)
+        {
+            var oldState = e.OldValue.ToString().ToLower();
+            var newState = e.NewValue.ToString().ToLower();
+
+            PseudoClasses.Remove($":{oldState}");
+            PseudoClasses.Add($":{newState}");
+
+            var state = (SplitViewDisplayMode)e.NewValue;
+            //PaneColumn Specs: Width/ColumnWidth
+            //   Overlay - 0px/0px
+            //   CompactOverlay - CompactPaneLength/CompactPaneLength
+            //   Inline - 0px/Auto
+            //   CompactInline - CompactPaneLength/Auto
+            TemplateSettings.ClosedPaneWidth = (state == SplitViewDisplayMode.CompactInline ||
+                state == SplitViewDisplayMode.CompactOverlay) ? CompactPaneLength : 0;
+
+            TemplateSettings.PaneColumnGridLength = (state == SplitViewDisplayMode.CompactInline ||
+                state == SplitViewDisplayMode.Inline) ? new GridLength(0, GridUnitType.Auto) :
+                state == SplitViewDisplayMode.Overlay ? new GridLength(0, GridUnitType.Pixel) :
+                new GridLength(CompactPaneLength, GridUnitType.Pixel);
```

Comment:
I think new switch syntax could fit here.

```suggestion
            var (closedPaneWidth, paneColumnGridLength) = (SplitViewDisplayMode)e.NewValue switch
            {
                SplitViewDisplayMode.Overlay => (0, new GridLength(0, GridUnitType.Pixel)),
                SplitViewDisplayMode.CompactOverlay => (CompactPaneLength, new GridLength(CompactPaneLength, GridUnitType.Pixel)),
                SplitViewDisplayMode.Inline => (0, new GridLength(0, GridUnitType.Auto)),
                SplitViewDisplayMode.CompactInline => (CompactPaneLength, new GridLength(0, GridUnitType.Auto)),
                _ => throw new NotImplementedException(),
            };
            TemplateSettings.ClosedPaneWidth = closedPaneWidth;
            TemplateSettings.PaneColumnGridLength = paneColumnGridLength;
```